### DISCUSSION TITLE
LOM204 (WISOL)

### DISCUSSION THREAD

**Discussion by tcpipchip** [hardwario/lora-modem-abz#113] [General]
Hi, 
Soon good news to WISOL USERS using your LIBRARIES :)

**Comment by tcpipchip** [comment]
[Datasheet_LOM204A_V0.1_20190715.pdf](https://github.com/hardwario/lora-modem-abz/files/9803648/Datasheet_LOM204A_V0.1_20190715.pdf)
Attached datasheet
Have you a good ABP AT seguence to me to follow and give a feedback ?

**Reply by janakj** [reply]
Does it need to be ABP? The easiest way to check if the module is working is with an OTAA Join.

**Reply by tcpipchip** [reply]
ok, i will do first!

**Reply by janakj** [reply]
First configure your [region](https://github.com/hardwario/lora-modem-abz/wiki/AT-Command-Interface#atband):
`AT+BAND=n`

Enable OTAA [mode](https://github.com/hardwario/lora-modem-abz/wiki/AT-Command-Interface#atmode):
`AT+MODE=1`

The modem uses the public network preamble by default. If your network uses the private preamble, switch it with [`AT+NWK`](https://github.com/hardwario/lora-modem-abz/wiki/AT-Command-Interface#atnwk).

Configure the device on the network server. Select LoRaWAN 1.0.4, RP2-1.0.3, OTAA mode. Use the default key (2B7E151628AED2A6ABF7158809CF4F3C) for all keys.

Send an [OTAA Join](https://github.com/hardwario/lora-modem-abz/wiki/AT-Command-Interface#atjoin):
`AT+JOIN`
If you get `+EVENT=1,1`, the device joined successfully. If you get `+EVENT=1,0`, the Join didn't work. It can take up to two minutes to get the event.

**Comment by tcpipchip** [comment]
![image](https://user-images.githubusercontent.com/11488460/196258699-eb08ea79-f16a-4080-ae63-dd29fced42bf.png)
Boot :)

**Comment by tcpipchip** [comment]
getting success on join
See at end of blog
https://lom204-cli-wisol.blogspot.com/2022/10/wisol-lom204-mais-uma-opcao-de-comandos.html
Now tryng to send a package, not reach the server
![image](https://user-images.githubusercontent.com/11488460/196408799-865935bb-7364-4c9f-be63-9f275b0f964a.png)

**Comment by tcpipchip** [comment]
got some success with
AT+PCTX 1,5\rhello
![image](https://user-images.githubusercontent.com/11488460/196414914-55c7bfc5-2748-4dc2-872f-70d624386409.png)


but not with 
AT+PUTX

**Reply by janakj** [reply]
Your uplinks seem to require too many retransmissions (`+EVENT=2,2`). This could be due to poor RF conditions, incorrect channel mask, or some other problem with the modem.

**Reply by tcpipchip** [reply]
> Your uplinks seem to require too many retransmissions (`+EVENT=2,2`). This could be due to poor RF conditions, incorrect channel mask, or some other problem with the modem.

i changed now to 1.0.4 specification...
Continuing the tests!

**Comment by tcpipchip** [comment]
getting progress!

**Comment by tcpipchip** [comment]
looks that sometime i have to send
AT+CHMASK=00FF00000000000000
before
AT+PCTX 1,5\rhello
or
AT+PUTX 1,5\rhello
when i get from them
+EVENT=2,2
+EVENT=2,2
+EVENT=2,2

Then works perfectly :(

**Reply by janakj** [reply]
In OTAA mode, the channel mask should have been set by the network server in Join response. In ABP mode, you may need to configure it yourself after device factory reset or band change.

**Reply by tcpipchip** [reply]
> In OTAA mode, the channel mask should have been set by the network server in Join response. In ABP mode, you may need to configure it yourself after device factory reset or band change.

AT+CHMASK=00FF00000000000000
will choose the channels 8 to 15, right ?

![image](https://user-images.githubusercontent.com/11488460/196444154-a61d42ed-74ee-4fe7-9de5-56b8f25fd232.png)

**Reply by janakj** [reply]
Yes, the above channel mask selects channels 8 through 15.

**Reply by tcpipchip** [reply]
> Yes, the above channel mask selects channels 8 through 15.

good!

**Comment by tcpipchip** [comment]
btw, your document is very very complete!

Plans to embedded a Script Language into module to be OPENCPU ?

**Reply by janakj** [reply]
No plans to add scripting to the modem itself. There is a [Python management tool](https://github.com/hardwario/lora-modem-abz/tree/main/python) that communicates with the modem via the AT command interface.

**Comment by tcpipchip** [comment]
ok!
Wich is the difference between AT+ and AT$ commands ?

**Reply by janakj** [reply]
From https://github.com/hardwario/lora-modem-abz/wiki/AT-Command-Interface:

> The modem implements two sets of AT commands. The commands that start with the prefix AT+ are designed to be compatible with the AT commands implemented by the Murata Modem firmware in the [Type ABZ-93](https://www.murata.com/en-us/products/connectivitymodule/lpwa/overview/lineup/type-abz-093) line of modules. The commands that start with the prefix AT$ are custom extension AT commands designed specifically for this firmware. The extension commands generally provide additional functionality not found in the AT+ set.

**Comment by tcpipchip** [comment]
wow, nice!

**Comment by tcpipchip** [comment]
One more question, this source if open-source, right ?
Then i can port to other modules too ?

**Reply by janakj** [reply]
Yes, it's open source and free.

**Comment by tcpipchip** [comment]
btw, i have plans to use you github to use in classroom in IoT, i am a teacher

Did you see the blog ?

**Comment by janakj** [comment]
Unfortunately, I don't speak Portuguese, sorry.

**Comment by tcpipchip** [comment]
looks that
AT+CHMASK=00FF00000000000000
is not storing in EEPROM...
because is backing to FFFFFF....without factory reset

**Reply by janakj** [reply]
The channel mask setting is stored in EEPROM. It reverts to the default value when you switch band. Also, the network server can change the value remotely, depending on how you have the network server configured.

**Reply by tcpipchip** [reply]
> The channel mask setting is stored in EEPROM. It reverts to the default value when you switch band. Also, the network server can change the value remotely, depending on how you have the network server configured.

thank you @janakj

**Comment by tcpipchip** [comment]
hi 
May i give a suggestion to use your lib as opencpu ?
Maybe is a easy for you to do...

**Reply by janakj** [reply]
What does opencpu mean?

**Comment by tcpipchip** [comment]
hi
means that the user can put inside his Application, to not need a host 
for example
My idea is to have a C function that simulate a AT comand  received to UART

send_at_cmd("AT+")
and 
receive_at_answer()

**Reply by janakj** [reply]
The firmware is open source. You can put anything you want in there and build your own binary.

**Comment by tcpipchip** [comment]
nice!

**Comment by tcpipchip** [comment]
i modifyng the code to make AT avaliable into the processor, without a HOST
what do you think about we add this feature ?

**Reply by janakj** [reply]
I am not sure I understand what you mean. Maybe you could rephrase or provide some examples?

**Comment by tcpipchip** [comment]
look
	enqueue((unsigned char *)"AT+VER?\r\n",9);
	HAL_Delay(3000);
It´s working, i can send an AT COMMAND without a HOST

My ideia is to continue, now reading the AT COMMAND answer without HOST

My opnion, it´s a big feature to add to make your LIB opencpu :)

What do you think ?

**Reply by janakj** [reply]
What would be the use case for this feature?

**Comment by tcpipchip** [comment]
mainly reduce hardware cost!

**Reply by janakj** [reply]
If you are trying to use the Type ABZ module as a standalone device, e.g., a sensor, then that's certainly possible, but you don't need to use the AT command interface from within the modem software itself. The AT command interface only makes sense when you need to communicate with the modem via a UART.

In your scenario, simply connect the sensor to a GPIO, access it through the STM32 HAL, and invoke LoRaMac-node functions directly, using the functions in src/cmd.c and src/lrw.c as an example.

**Comment by tcpipchip** [comment]
ok

**Comment by tcpipchip** [comment]
Can you provide a simple example as OPENCPU that only send a AT+VER? and get the results ?

By the way, here the  blog

https://lom204-cli-wisol.blogspot.com/2022/10/wisol-lom204-mais-uma-opcao-de-comandos.html

**Comment by tcpipchip** [comment]
hi again :)
Can you provide a simple example as OPENCPU ?
i have plans to insert on BLOG
